client/src/components/ResourceManager.jsx

```diff
@@ -2,14 +2,72 @@
 
 import React, { useEffect, useState } from 'react';
 import { Menu } from 'antd';
-import { DownOutlined, EditOutlined, CloseOutlined } from '@ant-design/icons';
+import {
+  DownOutlined,
+  UpOutlined,
+  EditOutlined,
+  CloseOutlined,
+  PlusOutlined,
+} from '@ant-design/icons';
 
 import { getCategories } from '../utils/api';
 
 import '../css/ResourceManager.css';
 
-const ResourceManager = () => {
+type Props = {
+  categories: { [string]: Array<string> },
+  categoryName: String,
+};
+
+const SidebarCategory = (props: Props) => {
   const { SubMenu } = Menu;
+
+  const { categories, categoryName } = props;
+  const [opened, setOpened] = useState(false);
+
+  const handleChange = () => {
+    setOpened(!opened);
+  };
+
+  return (
+    <SubMenu
+      {...props}
+      title={
+        <span>
+          {categoryName}
+          <span style={{ position: 'absolute', right: 15 }}>
+            <EditOutlined />
+            <CloseOutlined style={{ color: '#FF0000' }} />
+          </span>
+        </span>
+      }
+      icon={
+        opened ? (
+          <UpOutlined style={{ position: 'relative', top: -2 }} />
+        ) : (
+          <DownOutlined style={{ position: 'relative', top: -2 }} />
+        )
+      }
+      onTitleClick={handleChange}
+    >
+      {categories[categoryName].map((subCategory) => (
+        <Menu.Item key={subCategory} className="subcategory">
+          {subCategory}
+          <div>
+            <EditOutlined />
+            <CloseOutlined style={{ color: '#FF0000' }} />
+          </div>
+        </Menu.Item>
+      ))}
+      <Menu.Item key="Add Subcategory" className="subcategory">
```

Comment:
React uses keys to determine "uniqueness" - when I click on an "Add Subcategory" item it looks like React sees them as all the same thing.  We can probably get around this by making the key something like `add-subcategory-${categoryName}` (or really anything that just includes categoryName)

Reply:
Actually, I just thought about this - since Neha is making a map of category names to category IDs, when you merge with her, it might also be better to use the ID instead of the name